packages/python-packages/apiview-copilot/evals/_runner.py

```diff
@@ -68,24 +72,38 @@ def run(self):
         else:
             kwargs = {}
 
-        for file in self._test_files:
-            if not file.exists():
-                raise ValueError(f"Test file not found: {file}")
-            file_run_results = []
-            for run in range(self.num_runs):
-                print(f"Running evals {run + 1}/{self.num_runs} for {file.name}...")
-                result = evaluate(
-                    data=str(file),
-                    evaluators={"metrics": evaluator},
-                    evaluator_config={"metrics": evaluator.evaluator_config},
-                    target=evaluator.target_function,
-                    # FIXME: Should this be True? Probably?
-                    fail_on_evaluator_errors=False,
-                    # azure_ai_project=azure_ai_project,
-                    **kwargs,
-                )
-                file_run_results.append({file.name: result})
-            all_run_results.extend(file_run_results)
+        try:
+            for file in self._test_files:
+                if not file.exists():
+                    raise ValueError(f"Test file not found: {file}")
+
+                filtered_file_path = self._filter_jsonl_data(str(file))
+                if filtered_file_path != str(file):
+                    temp_files.append(filtered_file_path)
```

Comment:
I think a comment explaining the logic here could be helpful (I had to dig into `_filter_jsonl_data` a bit to understand the flow/intent)